taxon_filter.py

```diff
@@ -156,115 +156,47 @@ def bmtagger_wrapper(inBam, db, outBam, threads, JVMmemory=None):
 # =======================
 
 
-def lastal_chunked_fastq(
-    inFastq,
+def filter_lastal_bam(
+    inBam,
     db,
-    outFastq,
+    outBam,
     max_gapless_alignments_per_position=1,
     min_length_for_initial_matches=5,
     max_length_for_initial_matches=50,
     max_initial_matches_per_position=100,
-    chunk_size=100000
+    JVMmemory=None
 ):
+    ''' Restrict input reads to those that align to the given
+        reference database using LASTAL.
+    '''
 
-    lastal_path = tools.last.Lastal().install_and_get_path()
-    mafsort_path = tools.last.MafSort().install_and_get_path()
-    mafconvert_path = tools.last.MafConvert().install_and_get_path()
-    no_blast_like_hits_path = os.path.join(util.file.get_scripts_path(), 'noBlastLikeHits.py')
+    with util.file.tmp_dir('-lastal_db') as tmp_db_dir:
+        # auto build db if needed
+        if not all(os.path.exists(db + x)
+            for x in ('.bck', '.des', '.prj', '.sds', '.ssp', '.suf', '.tis')):
```

Comment:
Maybe the list of DB extensions should be part of `tools.last`?

Reply:
done!